src/Sulu/Bundle/AdminBundle/Resources/js/containers/WritingAssistant/Message.js

```diff
@@ -0,0 +1,190 @@
+// @flow
+
+import React, {Component, Fragment} from 'react';
+import {observable} from 'mobx';
+import {Button} from '../../components';
+import {TextEditor} from '../../containers';
+import {translate} from '../../utils';
+import messageStyles from './message.scss';
+
+type Props = {|
+    collapsed: boolean,
+    command: string,
+    displayActions: boolean,
+    expert: ?string,
+    index: number,
+    isLoading: boolean,
+    locale: string,
+    onClick?: (index: number) => void,
+    onCopy: (text: string) => void,
+    onInsert: (text: string) => void,
+    onRetry: (index: number) => void,
+    text: string,
+    title?: string,
+    type: 'text_line' | 'text_area' | 'text_editor',
+|};
+
+const MAX_LENGTH = 80;
+
+/**
+ * @internal
+ */
+class Message extends Component<Props> {
+    handleOnRetry = () => {
+        const {
+            index,
+            onRetry,
+        } = this.props;
+
+        onRetry(index);
+    };
+
+    handleOnCopy = () => {
+        const {
+            text,
+            onCopy,
+        } = this.props;
+
+        onCopy(text);
+    };
+
+    handleOnInsert = () => {
+        const {
+            text,
+            onInsert,
+        } = this.props;
+
+        onInsert(text);
+    };
+
+    handleOnClick = () => {
+        const {
+            index,
+            onClick,
+        } = this.props;
+        if (!onClick) {
+            return;
+        }
+
+        onClick(index);
+    };
+
+    trimText = (text: string) => {
+        // strip html tags
+        text = text?.replace(/<\/?[^>]+(>|$)/g, '') ?? '';
+        if (text.length <= 100) {
+            return text;
+        }
+
+        const firstPartEnd = text.lastIndexOf(' ', 70);
+        const firstPart = firstPartEnd !== -1 ? text.substring(0, firstPartEnd) : text.substring(0, 70);
+
+        const lastPartStart = text.slice(-20).indexOf(' ');
+        const lastPart = lastPartStart !== -1
+            ? text.substring(text.length - 20 + lastPartStart + 1)
+            : text.substring(text.length - 20);
+
+        return `${firstPart} ... ${lastPart}`;
+    };
+
+    renderTextComponent = () => {
+        const {
+            type,
+            text,
+            collapsed,
+            locale,
+        } = this.props;
+
+        if (collapsed) {
+            return this.trimText(text);
+        }
+
+        if (type === 'text_editor' && !collapsed) {
+            return (
+                <div className={messageStyles.textEditor}>
+                    <TextEditor
+                        adapter="ckeditor5"
+                        disabled={true}
+                        locale={observable.box(locale)}
+                        onChange={this.handleTextEditorChange}
+                        value={text}
+                    />
+                </div>
+            );
+        }
+
+        return text;
```

Comment:
It is always `!collapsed` as `collapsed` did early return.

I think best is to early return also for ! text editor to keep complex parts not inside if statement for readability:

```suggestion
        if (type !== 'text_editor') {
            return text;
        }

        return (
            <div className={messageStyles.textEditor}>
                <TextEditor
                    adapter="ckeditor5"
                    disabled={true}
                    locale={observable.box(locale)}
                    onChange={this.handleTextEditorChange}
                    value={text}
                />
            </div>
        );
```